src/libraries/System.Net.Http/src/System/Net/WebTransport/WebTransportSession.cs

```diff
@@ -0,0 +1,297 @@
+using System;
+using System.Threading.Tasks;
+using System.Threading;
+using System.IO;
+using System.IO.Ports;
+using System.Net.WebSockets;
+using System.Text;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace System.Net.WebTransport;
+
+// TODO: https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3-12#name-use-of-keying-material-expo
+// TODO: maybe use IAsyncDisposable instead of IDisposable everywhere?
+// TODO: add throw ArgumentNullException for all properties that are not nullable in all files
+
+public enum WebTransportSessionState
+{
+    None = 0;
+    /// <summary>
+    /// The connection is negotiating the handshake with the remote endpoint.
+    /// </summary>
+    Connecting;
+
+    /// <summary>
+    /// The initial state after the HTTP handshake has been completed.
+    /// </summary>
+    Open;
+
+    /// <summary>
+    /// A close message was sent to the remote endpoint.
+    /// </summary>
+    CloseSent;
```

Comment:
Is there an actual close handshake for the WT session? I thought it ends when the "control" stream is closed -- but maybe it's bidirectional and you refer to it being half-closed?

We might need to reword this and the following state to correspond to the actual behavior better?

NOTE: this can also be addressed later